api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -38,52 +47,372 @@ class Meta:
+                            begin=begin.isoformat(timespec="seconds"),
+                            end=end.isoformat(timespec="seconds"),
+                        )
+                    )
+                    continue
+
+                # Would the reservation be fully inside any reservable timespans for the resource?
+                # Ignores buffers for the reservation, since those can be outside reservable times.
+                if check_opening_hours and not any(
+                    timespan.fully_inside_of(reservable) for reservable in reservable_timespans
+                ):
+                    not_reservable.append(
+                        ReservationPeriodJSON(
+                            begin=begin.isoformat(timespec="seconds"),
+                            end=end.isoformat(timespec="seconds"),
+                        )
+                    )
+                    continue
+
+                non_overlapping.append(ReservationPeriod(begin=begin, end=end))
+
+        if overlapping:
+            msg = "Not all reservations can be made due to overlapping reservations."
+            extensions = {"code": error_codes.RESERVATION_SERIES_OVERLAPS, "overlapping": overlapping}
+            raise GraphQLError(msg, extensions=extensions)
+
+        if not_reservable:
+            msg = "Not all reservations can be made due to falling outside reservable times."
+            extensions = {"code": error_codes.RESERVATION_SERIES_NOT_OPEN, "not_reservable": not_reservable}
+            raise GraphQLError(msg, extensions=extensions)
+
+        # Pick out the through model for the many-to-many relationship and use if for bulk creation
+        ThroughModel: type[models.Model] = Reservation.reservation_unit.through  # noqa: N806
+
+        reservations: list[Reservation] = []
+        through_models: list[models.Model] = []
+
+        for period in non_overlapping:
+            if instance.reservation_unit.reservation_block_whole_day:
+                reservation_details["buffer_time_before"] = instance.reservation_unit.actions.get_actual_before_buffer(
+                    period["begin"]
+                )
+                reservation_details["buffer_time_after"] = instance.reservation_unit.actions.get_actual_after_buffer(
+                    period["end"]
+                )
+
+            reservation = Reservation(
+                begin=period["begin"],
+                end=period["end"],
+                recurring_reservation=instance,
+                age_group=instance.age_group,
+                **reservation_details,
+            )
+            through = ThroughModel(
+                reservation=reservation,
+                reservationunit=instance.reservation_unit,
+            )
+            reservations.append(reservation)
+            through_models.append(through)
+
+        Reservation.objects.bulk_create(reservations)
+        ThroughModel.objects.bulk_create(through_models)
+
+    def get_reserved_timespans(
```

Comment:
Consider using `Reservation.objects.get_affecting_reservations_as_closed_time_spans` and `merge_overlapping_time_span_elements` to replace this method.

Reply:
Yes, this simplifies things. Fixed.